mev_inspect/aave_liquidations.py

```diff
@@ -0,0 +1,130 @@
+    tx = []
+    liquidations = []
+    result = []
+    unique_transfer_hashes: List[str]
+    transfers_to: List[List] = [[]]
+    transfers_from: List[List] = [[]]
+    liquidator: str
+
+    for trace in traces:
+
+        if isinstance(trace, DecodedCallTrace):
+
+            # Check for liquidation and register trace and unique liquidator
+            if (
+                trace.classification == Classification.liquidate
+                and trace.transaction_hash not in tx
+            ):
+
+                liquidations.append(trace)
+                LIQUIDATION_CONTRACT_ADDRESSES.append(trace.from_address)
+                tx.append(trace.transaction_hash)
+
+                # Found liquidation, now parse inputs for data
+                for input in trace.inputs:
+
+                    if input == "_purchaseAmount":
+                        liquidation_amount = trace.inputs[input]
+                    elif input == "_collateral":
+                        collateral_address = trace.inputs[input]
+                        # This will be the address of the sent token
+                    elif input == "_reserve":
+                        reserve = trace.inputs[input]
+                        # This will be the address of the received token
+                    elif input == "_user":
+                        liquidated_user = trace.inputs[input]
+                # Register liquidation
+                result.append(
+                    Liquidation(
+                        liquidated_user=liquidated_user,
+                        collateral_address=collateral_address,
+                        collateral_amount=liquidation_amount,
+                        collateral_source="",
+                        reserve=reserve,
+                    )
+                )
+
+            elif is_transfer_from_liquidator(trace, unique_transfer_hashes):
+
+                # Add the transfer
+                liquidator = next(
+                    (
+                        LIQUIDATION_CONTRACT_ADDRESSES[i]
+                        for i in range(len(LIQUIDATION_CONTRACT_ADDRESSES))
+                        if trace.inputs["sender"] == LIQUIDATION_CONTRACT_ADDRESSES[i]
+                    ),
+                    "",
+                )
+
+                transfers_from.append(
+                    ["from", liquidator, trace.transaction_hash, trace.inputs["amount"]]
+                )
+                unique_transfer_hashes.append(trace.transaction_hash)
+
+            elif is_transfer_to_liquidator(trace):
+
+                # Add the transfer
+                liquidator = next(
+                    (
+                        LIQUIDATION_CONTRACT_ADDRESSES[i]
+                        for i in range(len(LIQUIDATION_CONTRACT_ADDRESSES))
+                        if trace.inputs["recipient"]
+                        == LIQUIDATION_CONTRACT_ADDRESSES[i]
+                    ),
+                    "",
```

Comment:
Rather than empty string this is better as `None` - then strings can be considered valid and you can explicitly handle the case of it being absent

Reply:
actually, a clearer way to write this would be:
```
recipient = trace.inputs["recipient"]
if recipient in LIQUIDATION_CONTRACT_ADDRESSES:
    # do whatever
else:
    # do whatever else
```